tests/test_pipelines.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+import json
+from datetime import datetime
+from StringIO import StringIO
+import subprocess as sp
+from scrapy.exceptions import DropItem
+from pa11ycrawler.pipelines import (
+    DuplicatesPipeline, DropDRFPipeline, Pa11yPipeline
+)
+
+
+def test_duplicates_pipeline():
+    dup_pl = DuplicatesPipeline()
+    spider = object()
+    # first item: no problem
+    item1 = {"url": "google.com"}
+    processed1 = dup_pl.process_item(item1, spider)
+    assert item1 == processed1
+
+    # second item is different, so no problem
+    item2 = {"url": "edx.org"}
+    processed2 = dup_pl.process_item(item2, spider)
+    assert item2 == processed2
+
+    # third is the same as a previous, so raises an exception
+    item3 = {"url": "google.com"}
+    with pytest.raises(DropItem):
```

Comment:
Unfortunately, I can't. This is due to the architecture of Scrapy, and probably requires some explanation.

We're using three different Scrapy concepts here: spiders, items, and pipelines. A spider's job is to start from a given URL (or list or URLs) and crawl links to get from page to page. Each page is parsed into an item, which represents structured data from the page that a pipeline can operate on. In this case, `EdxSpider` is crawling pages and turning each one into an `A11yItem`, which saves the URL, request headers, access time, and page title. Spiders continue crawling until they've hit every URL they can, or until they're given a signal to stop.

The items produced by spiders are put into a queue and processed by the pipelines. In this case, the `Pa11yPipeline` takes an item, runs `pa11y` with the given URL and headers, and saves the result into a file with the page title and access time. Spiders and pipelines operate completely independently of each other, and by the time a pipeline has started processing an item, the spider that produced the item has already moved on and started crawling other pages.

So because of this architecture, a pipeline can't tell a spider not to crawl a page. The most we can do is drop items that are duplicates of pages we've seen before, so we don't waste effort re-running `pa11y